mage_ai/frontend/components/ServerTimeDropdown/index.tsx

```diff
@@ -0,0 +1,157 @@
+  const [displayLocalServerTime, setDisplayLocalServerTime] = useState<boolean>(shouldDisplayLocalServerTime());
+  const [includeServerTimeSeconds, setIncludeServerTimeSeconds] = useState<boolean>(shouldIncludeServerTimeSeconds());
+  const [showDropdown, setShowDropdown] = useState<boolean>(false);
+  const [times, setTimes] = useState<Map<TimeZoneEnum, string>>(
+    currentTimes({ includeSeconds: true, timeZones: DISPLAYED_TIME_ZONES }),
+  );
+  const [top, setTop] = useState<number>(0);
+
+  const defaultTimeZone = displayLocalServerTime
+    ? TimeZoneEnum.LOCAL 
+    : TimeZoneEnum.UTC;
+
+  const isSmallBreakpoint = window.innerWidth < BREAKPOINT_MEDIUM;
+
+  const handleButtonClick = useCallback(() => {
+    setShowDropdown(prevState => !prevState);
+  }, []);
+  const setDropdownPosition = useCallback((top) => {
+    setTop(top);
+  }, []);
+  const updateTimes = useCallback(() => {
+    const updatedTimes = currentTimes({ 
+      includeSeconds: includeServerTimeSeconds, 
+      timeZones: DISPLAYED_TIME_ZONES, 
+    });
+    setTimes(prevState => {
+      if (prevState.size === updatedTimes.size 
+        && prevState.get(TimeZoneEnum.UTC) === updatedTimes.get(TimeZoneEnum.UTC)) {
+        return prevState;
+      }
+      
+      return updatedTimes;
+    });
+  }, [includeServerTimeSeconds]);
+
+  const toggleDisplayLocalServerTime = () => {
+    setDisplayLocalServerTime(storeDisplayLocalServerTime(!displayLocalServerTime));
+  };
+  const toggleIncludeServerTimeSeconds = () => {
+    setIncludeServerTimeSeconds(storeIncludeServerTimeSeconds(!includeServerTimeSeconds));
+  };
+
+  const toggleOptions = [
+    {
+      checked: displayLocalServerTime,
+      label: 'Show as local time',
+      onCheck: toggleDisplayLocalServerTime,
+    },
+    {
+      checked: includeServerTimeSeconds,
+      label: 'Include seconds',
+      onCheck: toggleIncludeServerTimeSeconds,
+    },
+  ];
+
+  useEffect(() => {
+    // Dynamically check the time every second and update the time display if necessary
+    const interval = setInterval(() => {
+      updateTimes();
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, [updateTimes]);
+
+  useEffect(() => {
+    // Immediately update the time display when "Include seconds" is toggled
+    updateTimes();
+  }, [includeServerTimeSeconds, updateTimes]);
+
+  if (!times) return null;
+
+  return (
+    <ClickOutside onClickOutside={() => setShowDropdown(false)} open>
+      <div style={{ position: 'relative' }}>
+        <ServerTimeButton 
+          disabled={isSmallBreakpoint}
+          mountedCallback={setDropdownPosition}
+          onClick={handleButtonClick}
+          time={times.get(defaultTimeZone)}
+          timeZone={abbreviatedTimezone(defaultTimeZone)}
+        />
+
+        {!isSmallBreakpoint && showDropdown && (
+          <DropdownContainerStyle top={top}>
+            <DropdownHeaderStyle>
+              <Text bold muted uppercase>Server Time</Text>
```

Comment:
I know we initially were thinking of `Server time` like in the designs, but does `Current time` feel more accurate? e.g. In the case of the server having some non-UTC and non-local timezone, "Server time" may not accurately reflect what the server time actually is.